tests/test_dataset.py

```diff
@@ -115,6 +115,14 @@ def testExistingOutput(self):
             if os.path.exists(testDir):
                 shutil.rmtree(testDir, ignore_errors=True)
 
+    def _checkOutputGen3(self, repo):
+        self.assertTrue(os.path.exists(repo), 'Output directory must exist.')
+        self.assertTrue(os.listdir(repo), 'Output directory must not be empty.')
+        self.assertTrue(os.path.exists(os.path.join(repo, 'butler.yaml')),
+                        'Output directory must have a butler.yaml file.')
+        butler = dafButler.Butler(repo)
```

Comment:
Won't this do the first three tests implicitly in that the Butler constructor would fail? Maybe comment here that you are being explicit in testing because you want to give a more compact reason for the problem?

Also note in a docstring for the method that the tests all require a local file Butler repo and won't work for the general case. (making this work generally is straightforward: use ButlerURI -- not worth it for now though)

Reply:
Does that really call for a comment? I think it's fairly standard practice to have assertions that test preconditions for later assertions, and some testing libraries provide explicit support for this (`unittest` doesn't).

I'll document that the code only supports directory-based repositories at the moment.

Reply:
My main thought is that two of the tests could be construed as internal implementation details for butler (files being present and there being a file called butler.yaml). All that really matters is that you can instantiate a butler with that root.